keps/1-kubernetes-enhancement-proposal-process.md

```diff
@@ -221,106 +218,34 @@ Metadata items:
-their roadmap; fleshed out with a design; started; and finally implemented. As
-Kubernetes continues to mature, hopefully metrics on the utilization of features
-will drive decisions on what features to maintain and which to deprecate and so
-it is possible that a KEP would be retired if its functionality no longer provides
-sufficient value to the community.
+A KEP is has the following states
+
+- `Draft`: The KEP has been proposed and is actively being defined.
+- `Deferred`: The KEP is proposed but not actively being worked on.
+- `Approved`: Using lazy consensus, the approvers have approved this KEP.
+  It will move forward to implementation.
+- `Rejected`: The approvers and authors have decided that this KEP is not moving forward.
+  The KEP is kept around as a historical document.
+- `Withdrawn`: The KEP has been withdrawn by the authors.
+- `Final`: The KEP has been implemented and is no longer actively changed.
+- `Replaced`: The KEP has been replaced by a new KEP.
+  The `superseded-by` metadata value should point to the new KEP.
 
 ### Git and GitHub Implementation
 
-Practically an KEP would be implemented as a pull request to a central repository
-with the following example structure
-
-```
-├── 0000-kep-template.md
-├── CODEOWNERS
-├── index.md
-├── sig-architecture
-│   ├── deferred
-│   ├── orphaned
-│   └── retired
-├── sig-network
-│   ├── deferred
-│   ├── kube-dns
-│   ├── orphaned
-│   └── retired
-├── sig-node
-│   ├── deferred
-│   ├── kubelet
-│   ├── orphaned
-│   └── retired
-├── sig-release
-│   ├── deferred
-│   ├── orphaned
-│   └── retired
-├── sig-storage
-│   ├── deferred
-│   ├── orphaned
-│   └── retired
-├── unsorted-to-be-used-by-newcomers-only
-└── wg-resource-management
-    ├── deferred
-    ├── orphaned
-    └── retired
-```
-
-where each SIG or working group is given a top level directory with subprojects
-maintained by the SIG listed in sub directories. For newcomers to the community
-an `unsorted-to-be-used-by-newcomers-only` directory may be used before an KEP
-can be properly routed to a SIG although hopefully if discussion for a potential
-KEP begins on the mailing lists proper routing information will be provided to
-the KEP author. Additionally a top level index of KEPs may be helpful for people
-looking for a complete list of KEPs. There should be basic CI to ensure that an
-`index.md` remains up to date.
-
-Ideally no work would begin within the repositories of the Kubernetes organization
-before a KEP has been approved by the responsible SIG or working group. While the
-details of how SIGs organize their work is beyond the scope of this proposal one
-possibility would be for each charter SIG to create a top level repository within
-the Kubernetes org where implementation issues managed by that SIG would be filed.
+KEPs are checked into the community repo under the `/kep` directory.
+KEPs can either be put at the top level of that directory or in a SIG specific subdirectory.
```

Comment:
I don't know if we want to do that yet, especially since we still need a way to generate these monotonically increasing numbers. I was personally envisioning the breakout in to separate directories to aid with automatically assigning SIGs to KEPs they need to review but since we're still bootstrapping this process I think we can have all KEPs land at the top level for now. The `Metadata` section should help us move them later

Reply:
I'm okay with just keeping things flat for now.

Reply:
One issue with flat directory here is that the OWNERS file here will be a bottleneck.  By having per-SIG directories we can shard this out a bit.

Reply:
Only as SIGs sign up for the process. I'm happy to add more people to the top level `OWNERS` file until then

Reply:
Sure -- SGTM.